traversal/planner/GraphPlanner.java

```diff
@@ -410,47 +385,35 @@ public String toString() {
         return str.toString();
     }
 
-    private class Initialiser {
-
-        private final LinkedHashSet<PlannerVertex<?>> queue;
-        private int edgeCount;
-
-        private Initialiser() {
-            queue = new LinkedHashSet<>();
-            edgeCount = 0;
+    private void setInitialValues() {
+        resetInitialValues();
+
+        PlannerVertex<?> start = vertices.values().stream().min(comparing(PlannerVertex::getCost)).get();
+        Set<PlannerVertex<?>> closedSet = new HashSet<>();
+        PriorityQueue<PlannerVertex<?>> open = new PriorityQueue<>(comparing(
+                v -> v.ins().stream().filter(e -> closedSet.contains(e.from())).mapToDouble(PlannerEdge.Directional::getCost).min().orElse(v.getCost())
+        ));
+        open.add(start);
+        int vertexOrder = 0;
+        while (!open.isEmpty()) {
+            PlannerVertex<?> v = open.poll();
+            assert !closedSet.contains(v);
+            closedSet.add(v);
+            v.setOrderInitial(vertexOrder++);
+            v.outs().stream().map(PlannerEdge.Directional::to).filter(x -> !closedSet.contains(x) && !open.contains(x))
+                    .forEach(open::add);
         }
```

Comment:
Greedy initialisation: start with the vertex with the smallest size. On each iteration, select the vertex from the open set with the smallest traversal cost, which is estimated as the minimum cost of all incoming edges. The neighbourhood of this vertex is then added to the open set for consideration. This is repeated until all vertices are in the closed set.